java/core/src/main/java/sleeper/core/statestore/SplitFileReferenceRequest.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2022-2024 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sleeper.core.statestore;
+
+import java.util.List;
+
+public class SplitFileReferenceRequest {
+    private final FileReference oldReference;
+    private final List<FileReference> newReferences;
+
+    private SplitFileReferenceRequest(FileReference oldReference, List<FileReference> newReferences) {
+        this.oldReference = oldReference;
+        this.newReferences = newReferences;
```

Comment:
Should it be possible to create a SplitFileReferenceRequest where the file name in oldReference is not the same as the file names in the newReferences? i.e. should we check this during the construction of the SplitFileReferenceRequest? If oldReference has a filename of file1 and newReferences.get(0).getFilename() is file2 and newReferences.get(1).getFilename() is file 3 does that make any sense?

Reply:
We probably don't want that to happen, but at the moment its not possible, as this constructor is private so the only way to create a `SplitFileReferenceRequest` is to use the static constructor, which uses `SplitFileReference.referenceForChildPartition()` for the new references